changelogs/unreleased/5657-shadialtarsha-minor.md

```diff
@@ -0,0 +1,42 @@
+## Add support for extensions.filters.http.ratelimit.v3.RateLimitPerRoute 
+
+Envoy has extensions.filters.http.ratelimit.v3.RateLimitPerRoute API which allows control over the Vhost Rate Limits on the route level.
+
+### Sample Configurations
+#### httpproxy.yaml
+```yaml
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: echo
+spec:
+  virtualhost:
+    fqdn: local.projectcontour.io
+    rateLimitPolicy:
+      global:
+        disabled: true
+      local:
+        requests: 100
+        unit: hour
+        burst: 20
+  routes:
+    - conditions:
+        - prefix: /
+      services:
+        - name: ingress-conformance-echo
+          port: 80
+    - conditions:
+        - prefix: /foo
+      rateLimitPolicy:
+        vhRateLimits: "Ignore"
```

Comment:
the name seems a bit awk to me here for some reason:

* `vhRateLimits`: is `vh` a common thing in the api?
* there is a different object in the CRD called `virtualhost` so this is a bit confusing

I think `routeRateLimitPolicy` (or just rateLimit) would be better imho

Reply:
I think here is you define the `merge` strategy actually, so I guess maybe something along that lines